### PR TITLE
feat: Implement design for settings modal

### LINKED ISSUE #438: Design for settings screen

### PR DESCRIPTION
**Pull request by sehyod**
This implements useful components (Checkbox, Input, ...) and the updated design for the Settings Modal.
The modal can now be closed by clicking outside of it or pressing Escape

Screenshot 2023-09-04 at 14 13 55

Closes #438

### CODE REVIEW

File: src/settings/SettingsModal.tsx [hunk 1]
```diff
@@ -47,55 +47,60 @@ const SETTINGS_PANES: SettingsPanesConfig[] = [
...
   return (
-    <div className="fixed left-0 top-0 z-modals flex h-screen w-screen items-center justify-center">
-      <div className="pointer-events-none absolute left-0 top-0 h-full w-full bg-slate-400/60" />
+    <div
+      className={cx(
+        'cursor-default select-none',
+        'fixed left-0 top-0 z-modals flex h-screen w-screen items-center justify-center',
+        'bg-modal-bg-overlay bg-opacity-[0.32]',
+      )}
+      onClick={onClose}
```
Comment (cguedes) [inline src/settings/SettingsModal.tsx:70]: You can use https://usehooks-ts.com/react-hook/use-on-click-outside

File: src/features/ai/settings/OpenAiSettingsPane.tsx [hunk 2]
```diff
@@ -43,79 +47,65 @@ export function OpenAiSettingsPane({ config }: SettingsPaneProps) {
...
+          onChange={(manner: RewriteMannerType) => setPaneSettings({ ...paneSettings, manner })}
+        />
+      </div>
+      <div className="flex flex-col items-start gap-2" ref={tooltipContainerRef}>
+        <div className="flex items-start gap-1 self-stretch">
+          <h2>Creativity Level</h2>
+          <CreativityInfoTooltip id="settings-creativity-tooltip" maxWidth={tooltipContainerRef.current?.clientWidth} />
+        </div>
+        <Slider
+          className="w-full"
```
Comment (cguedes) [inline src/features/ai/settings/OpenAiSettingsPane.tsx:99]: nit: I like to use an optional `fluid` prop for this in components.

File: src/features/ai/settings/OpenAiSettingsPane.tsx [hunk 3]
```diff
@@ -124,3 +114,14 @@ export function OpenAiSettingsPane({ config }: SettingsPaneProps) {
...
   );
 }
+
+function ChatModelTooltip() {
+  return (
+    <div className="text-btn-ico-tool-active">
+      <div id="chat-tooltip">
+        <InfoIcon />
+      </div>
+      {/* TODO: <Tooltip anchorSelect="#chat-tooltip" /> */}
```
Comment (cguedes) [inline src/features/ai/settings/OpenAiSettingsPane.tsx:124]: Should this block the merge?

**Review by cguedes — APPROVED** [review]

File: src/features/ai/settings/OpenAiSettingsPane.tsx (hunk 3 above)
Comment (sehyod) [reply src/features/ai/settings/OpenAiSettingsPane.tsx:124]: I think it's fine, I'll create a ticket to update the tooltip. The design team is aware of that and is working on the tooltip design and content

**Review by cguedes — APPROVED** [review]

**Pull request merged**